server/src/main/java/io/unitycatalog/server/persist/utils/RepositoryUtils.java

```diff
@@ -7,16 +7,30 @@
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
+import java.util.HashMap;
+import io.unitycatalog.server.utils.Constants;
 import org.hibernate.Session;
 
 public class RepositoryUtils {
+  private static final Map<String, Class<?>> PROPERTY_TYPE_MAP = new HashMap<>();
+  static {
+    PROPERTY_TYPE_MAP.put(Constants.FUNCTION, String.class);
+  }
   public static <T> T attachProperties(
       T entityInfo, String uuid, String entityType, Session session) {
     try {
       List<PropertyDAO> propertyDAOList =
           PropertyRepository.findProperties(session, UUID.fromString(uuid), entityType);
-      Method setPropertiesMethod = entityInfo.getClass().getMethod("setProperties", Map.class);
-      setPropertiesMethod.invoke(entityInfo, PropertyDAO.toMap(propertyDAOList));
+      Class<?> entityClass = PROPERTY_TYPE_MAP.getOrDefault(entityType, Map.class);
+      Method setPropertiesMethod =
+          entityInfo.getClass().getMethod("setProperties", entityClass);
+      Map<String, String> propertyMap = PropertyDAO.toMap(propertyDAOList);
+      Object propertiesArgument = switch (entityClass.getSimpleName()) {
+          case "Map" -> propertyMap;
+          case "String" -> propertyMap.toString();
```

Comment:
So, this function now uses json style string for functions and the map for others?

Reply:
Yes, because in `FunctionInfo` the argument for `setProperties` is a string unlike other entities (`Map<String, String>`). So, I took this approach to avoid "argument type mismatch".